core/src/solve.jl

```diff
@@ -763,28 +806,58 @@ function limit_flow!(
         )
     end
 
-    # UserDemand inflow is in [0, total_demand] and can be inactive
-    for (id, active) in zip(user_demand.node_id, user_demand.active)
-        total_demand = sum(
-            get_demand(user_demand, id, priority_idx, t) for
-            priority_idx in eachindex(allocation.priorities)
-        )
+    # UserDemand inflow bounds depend on multiple aspects of the simulation
+    for (id, active, inflow_edge, demand_from_timeseries) in zip(
+        user_demand.node_id,
+        user_demand.active,
+        user_demand.inflow_edge,
+        user_demand.demand_from_timeseries,
+    )
+        min_flow_rate, max_flow_rate = if demand_from_timeseries
+            # Bounding the flow rate if the demand comes from a time series is hard
+            0, Inf
+        else
+            inflow_id = inflow_edge.edge[1]
+            factor_basin_min =
+                min_low_storage_factor(current_storage, basin.storage_prev, inflow_id)
+            factor_level_min = min_low_user_demand_level_factor(
+                current_level,
+                basin.level_prev,
+                user_demand.min_level,
+                id,
+                inflow_id,
+            )
+            allocated_total =
+                is_active(allocation) ? sum(user_demand.allocated[id.idx, :]) :
+                sum(user_demand.demand[id.idx, :])
+            factor_basin_min * factor_level_min * allocated_total, allocated_total
+        end
         limit_flow!(
             u.user_demand_inflow,
             uprev.user_demand_inflow,
             id,
-            0.0,
-            total_demand,
+            min_flow_rate,
+            max_flow_rate,
             active,
             dt,
         )
     end
 
-    # Evaporation is in [0, ∞) (a stricter upper bound would require also estimating the area)
-    # Infiltration is in [0, infiltration]
+    # Evaporation is in [0, ∞) (stricter bounds would require also estimating the area)
+    # Infiltration is in [f * infiltration, infiltration] where f is a rough estimate of the smallest low storage factor
```

Comment:
There needs to be minimum infiltration? I thought the low storage could influence the maximum outflow, not the other way around?

Reply:
This limiter is not used to enforce the effect of the low storage factor. It does the opposite: it ensures that full infiltration is realised if enough storage is available